Documentation/Authentication.md

```diff
@@ -1,68 +1,127 @@
+All service APIs require authentication via a `credentials` object when being
+instantiated. There are three ways of authenticating and creating the required
+`credentials` via the SDK: basic authentication, interactive login, and service
+principal authentication.
 
-## 1. via Portal
-[This article](https://azure.microsoft.com/en-us/documentation/articles/resource-group-create-service-principal-portal/) provides detailed steps on creating a service principal via portal.
+## Basic Authentication
 
-## 2. via XplatCLI in 3 simple steps
-Pre-requisite:
-- Install the latest version of cli from the [latest github release](https://github.com/Azure/azure-xplat-cli/releases) or from [npm](https://npmjs.com/package/azure-cli).
+Simply provide your username and password to authenticate with the API using your
+Azure account. It is encouraged that your username and password be stored in
+environment variables rather than in the source code for your project.
 
-```bash
-npm uninstall azure-cli -g
-npm cache clear -g
-npm install azure-cli -g
+```js
+const Azure = require('azure');
+const MsRest = require('ms-rest-azure');
+
+MsRest.loginWithUsernamePassword(process.env.AZURE_USER, process.env.AZURE_PASS, (err, credentials) => {
+  if (err) throw err;
+
+  let storageClient = Azure.createARMStorageManagementClient(credentials, 'subscription-id');
+
+  // ..use the client instance to manage service resources.
+});
+```
+
+## Interactive login
+
+Interactive login will provide a link and a code that will allow the user to
+authenticate from a browser. Use this method when multiple accounts are used by
+the same script or when user intervention is preferred.
+
+```js
+const Azure = require('azure');
+const MsRest = require('ms-rest-azure');
+
+MsRest.interactiveLogin((err, credentials) => {
+  if (err) throw err;
+
+  let storageClient = Azure.createARMStorageManagementClient(credentials, 'subscription-id');
+
+  // ..use the client instance to manage service resources.
+});
 ```
-- Login to the azure-cli via azure login command to create a service principal for future use.
-  - For 2FA enabled user accounts  `azure login`
-  - For non 2FA enabled user accounts `azure login -u user1@constosocorp.com`
 
-After successful login, please follow the steps mentioned below to create a serviceprincipal:
-#### Step 0. Please make sure you have selected the correct subscription. The service principal will have access to your current subscription.
-- To list all your subscriptions: `azure account list`
-- To set the subscription of your choice: `azure account set "your subscription name"`
+## Service Principal Authentication
+
+Interactive login, similar to how the CLI authenticates, is the easiest way to
+authenticate; however, when using the Node.js SDK programmatically, you will need
+to use service principal authentication. This essentially creates keys for your
```

Comment:
> however, when using the Node.js SDK programmatically, you **will need to use service principal authentication**

It is not a requirement to use service principal authentication. The runtime supports all the 3 forms of authentication. The user may choose to use whatever seems fit for his scenario.